### PR TITLE
chore: Catch error for MTU Exchange

### PR DESCRIPTION
**Pull request by edwinavalos**
docs: add small trouble shooting section

Thanks for this project! Got it connected to my controller this evening after running into the silly error of looking at my controller on my phone while trying to connect!

I found that I was running into an issue with discovering services on the controller that corresponded to this `ATT request failed` error. I noticed that this error was actually happening earlier in the flow, first at the MTU Exchange, and noticed the error wasn't being caught. I added an error check there.

Very open to feedback on this! 

A note: I would love to extend this repo into something that can be used as a go single binary CLI for the controllers. I would think it'd be more appropriate as a separate repo, are you aware of any similar project in Go?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,3 +38,9 @@ Note that running the standalone binaries without root requires some extra capab
 ```sh
 sudo setcap 'cap_net_raw,cap_net_admin+eip' ./ac2mqtt
 ```
+
+### Troubleshooting
+
+**send ATT request failed: io: read/write on closed pipe, mac: xx:xx:xx:xx:xx:xx**
```
Comment (huguesb) [inline README.md:44]: is this still accurate if the MTU exchange error is caught?

**Comment by huguesb** [comment]
> Thanks for this project! Got it connected to my controller this evening after running into the silly error of looking at my controller on my phone while trying to connect!

Glad you're finding it useful. Yes, the controllers apparently only support once paired device at a time, it makes sense to add a note about that to the docs.

> I found that I was running into an issue with discovering services on the controller that corresponded to this `ATT request failed` error. I noticed that this error was actually happening earlier in the flow, first at the MTU Exchange, and noticed the error wasn't being caught. I added an error check there.

Thanks, I agree that it makes sense to log an error here.

> A note: I would love to extend this repo into something that can be used as a go single binary CLI for the controllers. I would think it'd be more appropriate as a separate repo, are you aware of any similar project in Go?

I am not aware of a similar project. Having a single binary CLI does seem like it could be useful and I agree that it's out of scope for this repo. If you want such a binary to operate a traditional UNIX-like command which spins up a new process for every command (as opposed to a single process with an interactive session) you will probably want to keep a daemon process in the background to hold the bluetooth connection otherwise latency and error rate are likely to go through the roof. I can think of basically three ways to go about it:
1. have your client talk to this bridge over an mqtt relay and submit patches to this repo to support more functionality as needed
2. fork this repo to have the client use a different IPC mechanism with the daemon process 
3. use bluez instead of raw hci sockets, which basically means talking to an existing daemon over D-Bus, I couldn't get it to work reliably in my case but your mileage might vary

I would happily welcome patches if you go with option 1. I would also potentially be open to accepting a patch for an alternate IPC channel to allow equivalent request/responses to be made via MQTT or another local IPC channel for use by a CLI depending on how much complexity that adds.

File: README.md (hunk 1 above)
Comment (edwinavalos) [reply README.md:44]: Yep, same error in both locations, initial connection is fine, but the rest of the operations will time out with this error.

**Comment by edwinavalos** [comment]
Thanks for your thoughts on the CLI project. I think I'll end up forking this and doing this base as the long running daemon, and doing method 2.

I actually don't use MQTT or Home Assistant, my curiosity has been mostly around reverse engineering the controller so that eventually I can have my own hardware controller, but have it be compatible with AC Infinity devices. Until I learn everything I will need to learn to be able to do that, I'll be hacking on this to try and get some better behavior around device shut off/turn on around chasing VPD. I don't think the controller software does a very good job at learning how long to run devices/how they approach things like Target VPD.

**Review by huguesb — APPROVED** [review]

**Pull request merged**